Jenkinsfile

```
@@ -1,97 +1,39 @@
-images = [
-  'centos7': [
-    'name': 'essdmscdm/centos7-build-node:3.0.0',
-    'sh': '/usr/bin/scl enable devtoolset-6 -- /bin/bash -e'
-  ],
-  'debian9': [
-    'name': 'essdmscdm/debian9-build-node:2.0.0',
-    'sh': 'bash -e'
-  ],
-  'ubuntu1804': [
-    'name': 'essdmscdm/ubuntu18.04-build-node:1.1.0',
-    'sh': 'bash -e'
-  ]
+container_build_bodes = [
+  'centos': ContainerBuildNode.getDefaultContainerBuildNode('centos7'),
+  'debian': ContainerBuildNode.getDefaultContainerBuildNode('debian9'),
+  'ubuntu': ContainerBuildNode.getDefaultContainerBuildNode('ubuntu1804')
 ]
 
-base_container_name = "${project}-${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
-
-def get_pipeline(image_key) {
-  return {
-    node('docker') {
-      def container_name = "${base_container_name}-${image_key}"
-      try {
-        def image = docker.image(images[image_key]['name'])
-        def custom_sh = images[image_key]['sh']
-        def container = image.run("\
-          --name ${container_name} \
-          --tty \
-          --cpus=2 \
-          --memory=4GB \
-          --network=host \
-          --env http_proxy=${env.http_proxy} \
-          --env https_proxy=${env.https_proxy} \
-          --env local_conan_server=${env.local_conan_server} \
-        ")
+packageBuilder = new ConanPackageBuilder(this, container_build_bodes, conan_pkg_channel)
+packageBuilder.defineRemoteUploadNode('centos')
 
-        stage("${image_key}: Checkout") {
-          sh """docker exec ${container_name} ${custom_sh} -c \"
-            git clone \
-              --branch ${env.BRANCH_NAME} \
-              https://github.com/ess-dmsc/${project}.git
-          \""""
-        }  // stage
+builders = packageBuilder.createPackageBuilders { container ->
+  packageBuilder.addConfiguration(container, [
+    'settings': [
+      'graylog_logger:build_type': 'Release'
+    ]
+  ])
+}
 
-        stage("${image_key}: Conan setup") {
-          withCredentials([
-            string(
-              credentialsId: 'local-conan-server-password',
-              variable: 'CONAN_PASSWORD'
-            )
-          ]) {
-            sh """docker exec ${container_name} ${custom_sh} -c \"
-              set +x
-              conan remote add \
-                --insert 0 \
-                ${conan_remote} ${local_conan_server}
-              conan user \
-                --password '${CONAN_PASSWORD}' \
-                --remote ${conan_remote} \
-                ${conan_user} \
-                > /dev/null
-            \""""
-          }  // withCredentials
-        }  // stage
+node {
+  checkout scm
 
-        stage("${image_key}: Package") {
-          sh """docker exec ${container_name} ${custom_sh} -c \"
-            cd ${project}
-            conan create . ${conan_user}/${conan_pkg_channel} \
-              --settings graylog-logger:build_type=Release \
-              --build=outdated
-          \""""
-        }  // stage
+  builders['macOS'] = get_macos_pipeline()
+  parallel builders
```

I am assuming that the *ecdc-pipeline* library has not changed much the past month or so. If it has not, we need some additional code for emailing build failures.
```suggestion
  try {
    parallel builders
    } catch (e) {
      pipeline_builder.handleFailureMessages()
      throw e
  }
```